flytekit/core/condition.py

```diff
@@ -176,8 +176,8 @@ def end_branch(self) -> Union[Condition, Promise]:
         """
         ctx = FlyteContextManager.current_context()
         # Let us mark the execution state as complete
-        ctx.execution_state.branch_complete()
-        if self._last_case:
+        ctx.execution_state.branch_complete()  # type: ignore
+        if self._last_case and self._selected_case:
             # We have completed the conditional section, lets pop off the branch context
             FlyteContextManager.pop_context()
             if self._selected_case.output_promise is None and self._selected_case.err is None:
```

Comment:
If we don't check it, this line might throw an exception

Reply:
oh right